sw/device/tests/pmod/i2c_host_eeprom_test.c

```diff
@@ -118,37 +116,37 @@ static status_t write_read_page(void) {
   return OK_STATUS();
 }
 
-static status_t test_init(void) {
-  mmio_region_t base_addr =
-      mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR);
-
-  TRY(dif_rv_core_ibex_init(base_addr, &rv_core_ibex));
-
-  base_addr = mmio_region_from_addr(TOP_EARLGREY_I2C2_BASE_ADDR);
-  TRY(dif_i2c_init(base_addr, &i2c));
+static status_t test_init(dif_i2c_t *i2c, dif_pinmux_t *pinmux,
+                          uintptr_t i2c_base_addr) {
+  mmio_region_t base_addr;
 
-  base_addr = mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR);
-  TRY(dif_pinmux_init(base_addr, &pinmux));
+  base_addr = mmio_region_from_addr(i2c_base_addr);
+  TRY(dif_i2c_init(base_addr, i2c));
 
-  TRY(i2c_testutils_select_pinmux(&pinmux, 2));
+  TRY(i2c_testutils_select_pinmux(pinmux, 2));
 
-  TRY(dif_i2c_host_set_enabled(&i2c, kDifToggleEnabled));
+  TRY(dif_i2c_host_set_enabled(i2c, kDifToggleEnabled));
 
   return OK_STATUS();
 }
 
 bool test_main(void) {
-  status_t test_result;
-  CHECK_STATUS_OK(test_init());
+  dif_pinmux_t pinmux;
+  dif_i2c_t i2c;
+
+  CHECK_DIF_OK(dif_pinmux_init(
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
```

Comment:
It's because the pinmux only needs to be initialized once, as opposed to i2c that needs to be initialized for every instance. So to avoid confusion, I've renamed the function name to i`2c_configure`